bindgen/lib.rs

```diff
@@ -2287,6 +2295,23 @@ fn ensure_libclang_is_loaded() {
 #[cfg(not(feature = "runtime"))]
 fn ensure_libclang_is_loaded() {}
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+enum GenerateError {
```

Comment:
Rather than an error this feels more like a result right? Whould something like:

```
enum GenerateResult {
  Ok(Bindings),
  ShouldRestart { ... },
  Err(BindgenError),
}
```

be a bit cleaner?